### PR TITLE
move-package: refactor `custom_resolve_pkg_name` to `custom_resolve_pkg_id` (pure refactor)

### PR DESCRIPTION
**Pull request by kklas**
## Description 

This is the famed rename from `name` to `identifier`, and boy am I glad we went for this because the code is 10x more readable now. In summary, it changes `*_pkg_name` -> `*_pkg_id` and `*_pkg_original_name` -> `*_pkg_name`.

In the second commit, I've found additional places where (what is now) `pkg_id` was being used in error messages instead of name so I switched those around. These don't show up in the unit tests because there aren't any that trigger those edge cases but it should be fine because it's clear from the code that these affect only error messages (in a good way) and not functionality of the resolver.

The only thing that remains is that we still use the `name = <...>` field in the lockfile for (what is now) `id`. I didn't want to touch this here because it would be a breaking change for the lockfile. In the next PR I'll be adding the additional fields to the lockfile that we discussed and switch this around.

This is part of the work to enable compiling against on-chain dependencies https://github.com/MystenLabs/sui/pull/14178.

cc @rvantonder @amnn

## Test Plan 

Unit tests

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: external-crates/move/crates/move-package/src/resolution/dependency_graph.rs [hunk 1]
```diff
@@ -66,21 +66,21 @@ use super::{
 pub struct DependencyGraph {
     /// Path to the root package and its name (according to its manifest)
     pub root_path: PathBuf,
-    pub root_package: PM::PackageName,
-    /// Root package name as defined in its manifest (can be different from the resolved name).
-    pub root_package_orig_name: PM::PackageName,
+    pub root_package_id: PackageIdentifier,
```
Comment (rvantonder) [inline external-crates/move/crates/move-package/src/resolution/dependency_graph.rs:69]: could we add a doc comment for this one?

**Review by rvantonder — APPROVED** [review]
looks great! nit, can we take this opportunity to describe the difference between package_name / package_id (I left an inline comment where I think we need this)

**Comment by kklas** [comment]
Good idea! I have added a docstring.

**Comment by rvantonder** [comment]
Thanks @kklas! set to automerge once CI is done

**Comment by kklas** [comment]
@rvantonder you might want to set automerge again as I pushed another small docstring fix

**Comment by rvantonder** [comment]
looks like it needs me to click approve again, so that stopped the automerge in any case. all good :-)

**Pull request merged**